scramjetorg/transform-hub
# Improve version endpoint to better identify running service

This PR improves `/version` endpoint response for host.

Previously it returned only `version` pointing to `package.json` version. Now it returns additional information to help better identify the service underneath:

```js
{
    "service": "host",   // service name
    "version": "0.19.2", // service version
    "apiVersion": "v1",  // service API version
    "build": "87fc6350"  // service build - git hash from which service is run
}
```

Git hash (`build`) is updated as a prebuild step in host package.

## How to test?

1. Start host:

```bash
yarn start     # build version
yarn start:dev # dev version
```

2. Query `/version` endpoint, for example:

```bash
curl localhost:8000/api/v1/version
```

It should return something like:

```
{"service":"host","apiVersion":"v1","version":"0.19.2","build":"87fc6350"}
```

## Naming

I'm not really convinced about naming, especially `host` (this is our official naming but may sound generic for someone not familiar with it) and `version` vs `apiVersion`. So any suggestions welcome.

---

If this is the direction we would like to go, I will update MW and potentially Manager, MM and MH in a similar way.
